cas: cas_test: create a real ACI.

sgotti opened the pull request.

kelseyhightower commented:
@sgotti There is a helper function for this now in https://github.com/coreos/rocket/blob/master/pkg/util/aci.go#L19. Can you rebase using the helper function?

sgotti commented:
@kelseyhightower in this and future tests I have the need to create a more complex ACI, with a custom imagemanifest and providing files inside the tar with different types and permissions. Plus I'd like to put the generated file in a specific tests dir. 

I added a commit that improves NewACI with these features and add a NewBasicACI with the previous behavior. Plus I added a dir argument to both functions.

Note: I noticed that appc/spec/aci.ArchiveWriter.AddFile has a path parameter that isn't used. After this PR, I'll open a PR to both appc/spec and rocket to remote it.

jonboulle commented on cas/cas_test.go:
```diff
@@ -50,8 +52,38 @@ func TestDownloading(t *testing.T) {
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	fmt.Printf("body: %s\n", body)
```
remove?

sgotti replied:
Whoops.

jonboulle commented on cas/cas_test.go:
```diff
@@ -50,8 +52,38 @@ func TestDownloading(t *testing.T) {
+
+	// Rewind the ACI
+	if _, err := aci.Seek(0, 0); err != nil {
+		t.Fatalf("unexpected error %v", err)
```
nit, should be `error:` for consistency

sgotti replied:
Done.

jonboulle commented on pkg/util/aci.go:
```diff
@@ -8,22 +8,34 @@ import (
-	manifest := fmt.Sprintf(`{"acKind":"ImageManifest","acVersion":"0.1.1","name":"%s"}`, name)
-
 	var im schema.ImageManifest
 	if err := im.UnmarshalJSON([]byte(manifest)); err != nil {
```
Check this before creating the tempfile.

sgotti replied:
Done.

jonboulle commented:
We could probably really just do this in memory but we can revisit another time. Change lgtm, couple of minor things to fix before landing.

jonboulle commented on pkg/util/aci.go:
```diff
@@ -31,6 +43,22 @@ func NewACI(name string) (*os.File, error) {
+			}
+		}
+		sr := strings.NewReader(entry.Contents)
+		if err := aw.AddFile("", entry.Header, sr); err != nil {
```
er, hmm, AddFile is clearly in need of an update since it does nothing with the first arg..

jonboulle commented:
thank you, lgtm
